### PR TITLE
Np 45158 Nvi point calculation: International collaboration

### PR DESCRIPTION
**Pull request by monaullah**
- International collaboration factor: check if any of the creators are affiliated with an institution with country code not equal to NO
- Also found and fixed another bug: `shouldNotCountCreatorSharesForNonCreators`. Added filter for contributors with role Creator in method `countCreatorShares()`

### CODE REVIEW

File: nvi-evaluator/src/main/java/no/sikt/nva/nvi/evaluator/calculator/PointCalculator.java [hunk 1]
```diff
@@ -24,13 +26,17 @@
...
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public final class PointCalculator {
 
     public static final String ERROR_MSG_EXTRACT_PUBLICATION_CONTEXT = "Could not extract publication channel for "
                                                                        + "instanceType {}, candidate: {}";
+    public static final String COUNTRY_CODE_NORWAY = "NO";
```
Comment (brinxmat) [inline nvi-evaluator/src/main/java/no/sikt/nva/nvi/evaluator/calculator/PointCalculator.java:38]: I have seen this in two PRs, possibly some kind of constant…possibly even from a standard library if ISO 3166-2 :)

File: nvi-evaluator/src/main/java/no/sikt/nva/nvi/evaluator/calculator/PointCalculator.java [hunk 2]
```diff
@@ -126,6 +129,23 @@ private static BigDecimal getInstanceTypeAndLevelPoints(String instanceType,
         return INSTANCE_TYPE_AND_LEVEL_POINT_MAP.get(instanceType).get(channelType).get(level);
     }
 
+    private static boolean isInternationalCollaboration(JsonNode jsonNode) {
+        return getJsonNodeStream(jsonNode, JSON_PTR_CONTRIBUTOR)
+                   .filter(PointCalculator::isCreator)
+                   .flatMap(contributorNode -> getJsonNodeStream(contributorNode, JSON_PTR_AFFILIATIONS))
```
Comment (brinxmat) [inline nvi-evaluator/src/main/java/no/sikt/nva/nvi/evaluator/calculator/PointCalculator.java:135]: Probably for both map ops. extract a named method to sy. what is being done here, so `.flatMap(this::extractAffiliation).map(this::extractCountryCode)` or a contraction of these

File: nvi-evaluator/src/test/java/no/sikt/nva/nvi/evaluator/EvaluateNviNviCandidateHandlerTest.java [hunk 3]
```diff
@@ -157,7 +158,8 @@ void shouldCreateNewCandidateEventOnValidAcademicArticle() throws IOException {
         var message = sentMessages.get(0);
         var messageBody =
             attempt(() -> objectMapper.readValue(message.messageBody(), CandidateEvaluatedMessage.class)).orElseThrow();
-        var expectedEvaluatedMessage = getExpectedEvaluatedMessage("AcademicArticle", BigDecimal.valueOf(0.7071),
+        var expectedEvaluatedMessage = getExpectedEvaluatedMessage("AcademicArticle",
+                                                                   BigDecimal.valueOf(1).setScale(SCALE, ROUNDING_MODE),
```
Comment (brinxmat) [inline nvi-evaluator/src/test/java/no/sikt/nva/nvi/evaluator/EvaluateNviNviCandidateHandlerTest.java:162]: This is getting repeated…probably extract a named method

File: nvi-evaluator/src/test/java/no/sikt/nva/nvi/evaluator/calculator/PointCalculatorTest.java [hunk 4]
```diff
@@ -15,13 +14,19 @@
...
 import java.util.stream.IntStream;
 import java.util.stream.Stream;
+import org.junit.jupiter.api.Test;
 import org.junit.jupiter.params.ParameterizedTest;
 import org.junit.jupiter.params.provider.MethodSource;
 
 class PointCalculatorTest {
 
+    public static final String COUNTRY_CODE_NO = "NO";
+    public static final String SOME_INTERNATIONAL_COUNTRY_CODE = "GB";
```
Comment (brinxmat) [inline nvi-evaluator/src/test/java/no/sikt/nva/nvi/evaluator/calculator/PointCalculatorTest.java:27]: Again…

Comment (brinxmat) [reply nvi-evaluator/src/test/java/no/sikt/nva/nvi/evaluator/calculator/PointCalculatorTest.java:27]: Probably the ISO 3166 code from `Locale`

File: nvi-evaluator/src/test/java/no/sikt/nva/nvi/evaluator/calculator/PointCalculatorTest.java [hunk 5]
```diff
@@ -115,18 +170,16 @@ private static Stream<PointParameters> twoCreatorsAffiliatedWithTwoDifferentInst
...
             new PointParameters("AcademicMonograph", "Series", "2", false, 4, bd("5.6569"), bd("4.0000")),
-            //new PointParameters("AcademicChapter", "Series", "1", true, 5, bd("0.6325"), bd("0.4472")),
+            new PointParameters("AcademicChapter", "Series", "1", true, 5, bd("0.8222"), bd("0.5814")),
             new PointParameters("AcademicChapter", "Series", "2", false, 5, bd("1.8974"), bd("1.3416")),
             new PointParameters("AcademicArticle", "Journal", "1", false, 7, bd("0.5345"), bd("0.3780")),
-            new PointParameters("AcademicArticle", "Journal", "2", false, 7, bd("1.6036"), bd("1.1339"))
-            //new PointParameters("AcademicLiteratureReview", "Journal", "1", true, 8, bd("0.6500"), bd("0.4596")),
-            //new PointParameters("AcademicLiteratureReview", "Journal", "2", true, 8, bd("1.9500"), bd("1.3789"))
+            new PointParameters("AcademicArticle", "Journal", "2", false, 7, bd("1.6036"), bd("1.1339")),
+            new PointParameters("AcademicLiteratureReview", "Journal", "1", true, 8, bd("0.6500"), bd("0.4596")),
```
Comment (brinxmat) [inline nvi-evaluator/src/test/java/no/sikt/nva/nvi/evaluator/calculator/PointCalculatorTest.java:181]: bd -> asBigDecimal or something

**Review by brinxmat — CHANGES_REQUESTED** [review]

File: nvi-evaluator/src/main/java/no/sikt/nva/nvi/evaluator/calculator/PointCalculator.java (hunk 1 above)
Comment (monaullah) [reply nvi-evaluator/src/main/java/no/sikt/nva/nvi/evaluator/calculator/PointCalculator.java:38]: I haven't found a standard library with Norways country code (with standard ISO 3166-2). java.util.Locale does not include Norway i think. Do you know of one? Should i put it in NVA commons?

**Review by hackon — APPROVED** [review]

**Pull request merged**